modules/perception/camera/app/cipv_camera.cc

```diff
@@ -0,0 +1,967 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/perception/camera/app/cipv_camera.h"
+
+#include <cmath>
+#include <limits>
+
+#include "cyber/common/log.h"
+#include "modules/common/math/line_segment2d.h"
+
+namespace apollo {
+namespace perception {
+
+Cipv::Cipv(void) {}
+
+Cipv::~Cipv(void) {}
+
+bool Cipv::Init(const Eigen::Matrix3d &homography_im2car) {
+  b_image_based_cipv_ = false;
+  debug_level_ = 0;  // 0: no debug message
```

Comment:
if we really want this flag, please make it as gflag so that it can be configured

Reply:
I changed it b_image_based_cipv_ configurable parameter.